### PR TITLE
bind unbind test for virtual device

### PR DESCRIPTION
**Pull request by smruti77**
remove vnic bind unbind test and added in to the the common
virtual bind unbind for vnic,veth,vscsi and veth device.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[veth_bind_unbind.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4544501/veth_bind_unbind.log)
[vfc_bind_unbind.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4544502/vfc_bind_unbind.log)
[vnic_bind_unbind.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4544503/vnic_bind_unbind.log)

File: io/common/virtual_bind_unbind.py.data/README.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+This script will unbind and bind the virtualized device
+for vnic, vscsi, vfc,veth as many times as specified by 
+the user
+
+Inputs Needed :
+------------------------------------
+virtual_device - virtual device for the test interface or disk name.
+virtual_slot - slot name for the virtual device, which can get it from lscfg -vl command.
```
Comment (abdhaleegit) [inline io/common/virtual_bind_unbind.py.data/README.txt:8]: update the README to get slot for vfc use lsslot, as vfc do not give correct slot number using lscfg -vl ( all other vdevices gives correct slot using lscfg)

```
[root@ltc-zz1b-lp1 ~]# lscfg -vl sdm
  sg4 1:0:1:0 sdm  **H1-B0-T1-L0**           16 BitVirtual I/O Device - SCSI
                                         Disk Drive
        Manufacturer Name.........IBM
        Machine Type-Model........2145
        Alterable ROM Level.......0000
        SCSI Host.(XH)............1
        SCSI Bus.(XB).............0
        SCSI Target.(XT)..........1
        SCSI Lun.(XL).............0
        First 8 bytes of SCSI query response.(Z0)
        ..........................00000007008083B0
        Device Specific.(Z7)......60050768028211D8
        Location Code.(YL)........H1-B0-T1-L0

[root@ltc-zz1b-lp1 ~]# lsslot
# Slot                     Description       Linux Name    Device(s)
U9009.42A.13C6F1W-V2-C0    Virtual I/O Slot  30000000      vty
U9009.42A.13C6F1W-V2-C2    Virtual I/O Slot  30000002      l-lan
U9009.42A.13C6F1W-V2-C201  Virtual I/O Slot  300000c9      vfc-client
```

Comment (abdhaleegit) [inline io/common/virtual_bind_unbind.py.data/README.txt:8]: also provide example slot name in README like
virtual_slot: U9009.42A.13C6F1W-V2-C201

File: io/common/virtual_bind_unbind.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        """
+        smm = SoftwareManager()
+        for pkg in ["net-tools"]:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        self.virtual_device = self.params.get('virtual_device')
+        self.virtual_slot = self.params.get('virtual_slot')
+        if "T" in self.virtual_slot:
+            self.virtual_slot = self.virtual_slot.split("-T")[0]
```
Comment (abdhaleegit) [inline io/common/virtual_bind_unbind.py:51]: why do you need this ?  the user will be giving input without -T
virtual_slot: U9009.42A.13C6F1W-V2-C3

so no need to remove it

Comment (smruti77) [reply io/common/virtual_bind_unbind.py:51]: ok

File: io/common/virtual_bind_unbind.py.data/README.txt (hunk 1 above)
Comment (smruti77) [reply io/common/virtual_bind_unbind.py.data/README.txt:8]: ok

File: io/common/virtual_bind_unbind.py (hunk 2 above)
Comment (abdhaleegit) [reply io/common/virtual_bind_unbind.py:51]: @smruti77 I got you, when a user finds slot using lscfg so he will have -T appended..  no issue you can keep for such cases

File: io/common/virtual_bind_unbind.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Bismruti Bidhibrata Pattjoshi<bbidhibr@in.ibm.com>
```
Comment (narasimhan-v) [inline io/common/virtual_bind_unbind.py:15]: Please add original author Harsha also

Comment (smruti77) [reply io/common/virtual_bind_unbind.py:15]: ok

**Comment by narasimhan-v** [comment]
@abdhaleegit re-review please

**Comment by abdhaleegit** [comment]
@narasimhan-v If a test involves more than one author , than it can have multiple  sign-off  rite ?

File: io/common/virtual_bind_unbind.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    This test verifies that for a given virtualized device.
+    :param device: Name of the virtualized device
+    """
+
+    def setUp(self):
+        """
+        Identify the virtualized device.
+        """
+        smm = SoftwareManager()
+        for pkg in ["net-tools"]:
```
Comment (abdhaleegit) [inline io/common/virtual_bind_unbind.py:45]: @smruti77  is this package name common to all distros ?

Comment (smruti77) [reply io/common/virtual_bind_unbind.py:45]: previously also it was common for all distro  i kept like this.

**Comment by abdhaleegit** [comment]
@smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine

**Comment by smruti77** [comment]
> @smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine

@abdhaleegit can't able to run this test because of ' -j ' option is not supported by sles.

**Comment by abdhaleegit** [comment]
> > @smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine
> 
> @abdhaleegit can't able to run this test because of ' -j ' option is not supported by sles.

What do you mean  by -j  , I don't see this option in script

**Comment by smruti77** [comment]
> > > @smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine
> > 
> > 
> > @abdhaleegit can't able to run this test because of ' -j ' option is not supported by sles.
> 
> What do you mean by -j , I don't see this option in script

@abdhaleegit its not in script its in new interfcae.py module.

**Comment by abdhaleegit** [comment]
> > > > @smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine
> > > 
> > > 
> > > @abdhaleegit can't able to run this test because of ' -j ' option is not supported by sles.
> > 
> > 
> > What do you mean by -j , I don't see this option in script
> 
> @abdhaleegit its not in script its in new interfcae.py module.

Any issue raised for the same ? in avocado ?

**Comment by smruti77** [comment]
> > > > > @smruti77 this scripts was run on RHEL distro, can you please have one run log on SLES, vNIC is fine
> > > > 
> > > > 
> > > > @abdhaleegit can't able to run this test because of ' -j ' option is not supported by sles.
> > > 
> > > 
> > > What do you mean by -j , I don't see this option in script
> > 
> > 
> > @abdhaleegit its not in script its in new interfcae.py module.
> 
> Any issue raised for the same ? in avocado ?

i told to narshiman

**Comment by smruti77** [comment]
[bind_unbind_vfc_sles.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4590971/bind_unbind_vfc_sles.log)
[bind_unbind_vnic_sles.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4590972/bind_unbind_vnic_sles.log)

**Comment by narasimhan-v** [comment]
@smruti77 So, was this run on latest SLES ?
Do we still need to worry about `ethtool -j` ?

**Comment by smruti77** [comment]
> @smruti77 So, was this run on latest SLES ?
> Do we still need to worry about `ethtool -j` ?

**Comment by smruti77** [comment]
> @smruti77 So, was this run on latest SLES ?
> Do we still need to worry about `ethtool -j` ?

that one  ip '-j' option i run it in sles15 sp2,it is working fine.

**Comment by narasimhan-v** [comment]
Thanks @smruti77 
@abdhaleegit review please

**Pull request merged**